src/ILLink.Shared/DiagnosticId.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace ILLink.Shared
+{
+	public enum DiagnosticId
+	{
+		// Linker diagnostic ids.
+		RequiresUnreferencedCode = 2026,
+		RequiresAttributeMismatch = 2046,
```

Comment:
Nit: Maybe is better to rename this to RequiresUnreferencedCodeAttributeMistmatch to match the RequiresAssemblyFilesAttributeMismatch format and be more verbose about the difference
```suggestion
		RequiresUnreferencedCodeAttributeMismatch = 2046,
```

Reply:
Changed it together we the warning message, PTAL.